### ISSUE TITLE
[Airbrake] [Production] invalid byte sequence in UTF-8

### ISSUE THREAD

**Issue by ryanfb** [sosol/sosol#165]
**Airbrake error:** [#1051](https://sosol.airbrake.io/projects/11795/groups/1665029452433801051)
**Airbrake project:** Papyrological Editor

**Error type:** `ArgumentError`
**Error message:** `invalid byte sequence in UTF-8`
**Where:** `<no information>`
**Occurred at:** Apr 18, 2016 19:50:53 UTC
**First seen at:** Apr 16, 2016 07:16:17 UTC
**Occurrences:** 3 (0 since last deploy on `<no information>`)

**URL:** [http://papyri.info/editor/Annam%E1ria-IzabellaP%E1zsint](http://papyri.info/editor/Annam%E1ria-IzabellaP%E1zsint)
**File:** `org/jruby/RubyString.java`

**Backtrace:**

```
org/jruby/RubyString.java:3101:in gsub
org/jruby/RubyString.java:3081:in gsub!
gems/gems/activesupport-3.2.18/lib/active_support/inflector/methods.rb:79:in underscore
gems/gems/activesupport-3.2.18/lib/active_support/core_ext/string/inflections.rb:106:in underscore
```

**Comment by ryanfb** [comment]
`%E1` is Windows-1252 encoded `á`. Not sure if this is because a URL copy/paste got messed up somewhere, but maybe we should double check that we explicitly use UTF-8 encoding/escaping when presenting user URLs. Also potentially recover from this more gracefully.

**Comment by ryanfb** [comment]
http://papyri.info/editor/all_users_links presents the link as http://papyri.info/editor/users/Annam%C3%A1ria-IzabellaP%C3%A1zsint and is UTF-8 encoded. Not sure what more we can do, aside from more gracefully handling, which would be a chore.

**Comment by ryanfb** [comment]
Still seeing tons of these errors. Apparently by adding some Rack middleware, we might be able to solve the problem: https://stackoverflow.com/questions/24648206/ruby-on-rails-invalid-byte-sequence-in-utf-8-due-to-bot/24727310

This is bundled up in a Gem/Engine here: https://github.com/sunny/handle_invalid_percent_encoding_requests

May be fixed in Rails 4.2 (see #209).

**Comment by ryanfb** [comment]
Closing with this live on production. Will reopen/revisit if the error comes up again.
